photon-lib/src/main/java/org/photonvision/estimation/VisionEstimation.java

```diff
@@ -55,20 +55,24 @@ public static List<AprilTag> getVisibleLayoutTags(
     }
 
     /**
-     * Performs solvePNP using 3d-2d point correspondences to estimate the field-to-camera
-     * transformation. If only one tag is visible, the result may have an alternate solution.
+     * Performs solvePNP using 3d-2d point correspondences of visible AprilTags to estimate the
+     * field-to-camera transformation. If only one tag is visible, the result may have an alternate
+     * solution.
      *
      * <p><b>Note:</b> The returned transformation is from the field origin to the camera pose!
-     * (Unless you only feed this one tag??)
+     *
+     * <p>With only one tag: {@link OpenCVHelp#solvePNP_SQUARE}
+     *
+     * <p>With multiple tags: {@link OpenCVHelp#solvePNP_SQPNP}
      *
      * @param cameraMatrix The camera intrinsics matrix in standard opencv form
      * @param distCoeffs The camera distortion matrix in standard opencv form
      * @param visTags The visible tags reported by PV
      * @param tagLayout The known tag layout on the field
-     * @return The transformation that maps the field origin to the camera pose
+     * @return The transformation that maps the field origin to the camera pose, or an empty Optional
+     *     if estimation fails.
      */
-    @Deprecated
-    public static PNPResults estimateCamPosePNP(
+    public static Optional<PNPResults> estimateCamPosePNP(
```

Comment:
The reason I split these functions is that the behavior the origin of the Pnpresult transform was based on if it sees 1 target or more than one, previously. Is that no longer the case? It looks like still if you see one tag it puts the tag at the origin, not at where it is on the field?

Reply:
There was a bug in `VisionEstimation` previously where that method would give `kTagModel.getFieldVertices(knownTags.get(0).pose)` to solvePNP_SQUARE instead of `TargetModel.kTag16h5.vertices` (which is required for that solvepnp method and is assumed in the transforms following it). It should now correctly return the origin-to-camera transform for both single and multi tag.

Reply:
Is there a unit test proving this?